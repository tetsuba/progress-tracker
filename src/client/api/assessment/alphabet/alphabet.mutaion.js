import gql from 'graphql-tag'

export const SAVE_ALPHABET_ASSESSMENT_MUTATION = gql`
  mutation SaveAlphabetAssessment($input: AlphabetAssessmentInput!) {
    saveAlphabetAssessment(input: $input) {
      studentName
      history {
        date
        alphabet {
          letter
          value
        }
      }
    }
  }
`
