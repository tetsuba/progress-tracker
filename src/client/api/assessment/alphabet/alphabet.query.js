import gql from 'graphql-tag'

export const GET_ALPHABET_ASSESSMENT = gql`
  query($input: GetAlphabetAsessmentInput!) {
    getAlphabetAssessment(input: $input) {
      studentName
      history {
        date
        alphabet {
          value
          letter
        }
      }
    }
  }
`
