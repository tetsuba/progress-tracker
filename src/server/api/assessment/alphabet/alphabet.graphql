
type AlphabetType {
  letter: String,
  value: Int,
}

type AlphabetHistoryType {
  date: String,
  alphabet: [AlphabetType]
}

type AlphabetAssessmentType {
  studentName: String
  history: [AlphabetHistoryType]!
}

input AlphabetInput {
  letter: String,
  value: Int,
}

input AlphabetAssessmentInput {
  alphabet: [AlphabetInput]
  date: String,
  studentId: String,
  typeOfAlphabetAssessment: String,
}

input GetAlphabetAsessmentInput {
  studentId: String,
  typeOfAlphabetAssessment: String,
}

extend type Query {
  getAlphabetAssessment(input: GetAlphabetAsessmentInput!): AlphabetAssessmentType!
}

extend type Mutation {
  saveAlphabetAssessment(input: AlphabetAssessmentInput!): AlphabetAssessmentType!
}
