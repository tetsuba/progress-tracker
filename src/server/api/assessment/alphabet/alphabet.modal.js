const mongoose = require('mongoose')

const AlphabetSchema = new mongoose.Schema({
  letter: {
    type: String,
  },
  value: {
    type: Number,
  },
})

const AlphabetHistorySchema = new mongoose.Schema({
  date: {
    type: String,
    default: '',
  },
  alphabet: {
    type: [AlphabetSchema],
    default: [],
  },
})

const AlphabetAssessmentSchema = new mongoose.Schema({
  studentId: {
    type: String,
  },

  typeOfAlphabetAssessment: {
    type: String,
    required: true,
  },

  history: {
    type: [AlphabetHistorySchema],
  },
})

module.exports = AlphabetAssessmentSchema
