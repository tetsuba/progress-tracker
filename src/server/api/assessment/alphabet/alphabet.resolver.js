const { errorName } = require('../../../errorHandling')
module.exports = {
  Query: {
    getAlphabetAssessment: async (_, args, context) => {
      const { Student, AlphabetAssessment } = context.models
      const { studentId, typeOfAlphabetAssessment } = args.input

      const student = await Student.findById(studentId)
      const assessment = await AlphabetAssessment.findOne({
        studentId: studentId,
        typeOfAlphabetAssessment: typeOfAlphabetAssessment,
      })

      return {
        studentName: student.firstName,
        history: assessment.history,
      }
    },
  },

  Mutation: {
    saveAlphabetAssessment: async (_, args, context) => {
      const { AlphabetAssessment, Student } = context.models
      const { studentId, alphabet, date, typeOfAlphabetAssessment } = args.input

      const student = await Student.findById(studentId)
      if (!student) {
        throw new Error(errorName.STUDENT_NOT_FOUND)
      }

      const assessment = await AlphabetAssessment.findOne({
        studentId: studentId,
        typeOfAlphabetAssessment: typeOfAlphabetAssessment,
      })

      if (!assessment) {
        throw new Error(errorName.ASSESSMENT_NOT_FOUND)
      }

      assessment.history.push({ alphabet: alphabet, date: date })
      await assessment.save()

      return {
        studentName: student.firstName,
        history: assessment.history,
      }
    },
  },
}
