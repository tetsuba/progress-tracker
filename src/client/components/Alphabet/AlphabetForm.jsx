import React, { useEffect, useState } from 'react'
import { useMutation } from '@apollo/react-hooks'
import { Button } from 'react-bootstrap'

// CONTEXT
import { LoadingContext } from '../../context/LoadingContext'

// COMPONENTS
import Alphabet from './Alphabet'

// QUERIES
import { GET_ALPHABET_ASSESSMENT } from '../../api/assessment/alphabet/alphabet.query'

// MUTATIONS
import { SAVE_ALPHABET_ASSESSMENT_MUTATION } from '../../api/assessment/alphabet/alphabet.mutaion'

// DEFAULT STATE
import ALPHABET_DEFAULT_STATE from './Alphabet.defualtState'
import { formatDefaultState } from './alphabet-utils'

// TODO: add loading indicator
type Props = {
  id: string,
  toggleModal: () => void,
  data: any,
  type: string,
}

export default function AlphabetForm(props: Props) {
  const { toggleModal, id, data, type } = props
  const defaultState =
    formatDefaultState(data) ||
    (type === 'lowerCase'
      ? ALPHABET_DEFAULT_STATE.map((state) => ({
          ...state,
          letter: state.letter.toLocaleLowerCase(),
        }))
      : ALPHABET_DEFAULT_STATE)

  const variables = {
    input: {
      studentId: id || '',
      typeOfAlphabetAssessment: type || '',
    },
  }

  const { showLoading, hideLoading } = React.useContext(LoadingContext)

  const [letters, setLetters] = useState(defaultState)
  const [saveAlphabetAssessment] = useMutation(
    SAVE_ALPHABET_ASSESSMENT_MUTATION,
    {
      refetchQueries: [{ query: GET_ALPHABET_ASSESSMENT, variables }],
    }
  )

  useEffect(() => {
    setLetters(defaultState)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <>
      <Alphabet letters={letters} update={setLetters} />
      <Button
        className="float-right"
        onClick={() => {
          const variables = {
            input: {
              studentId: id,
              date: String(Date.now()),
              alphabet: letters,
              typeOfAlphabetAssessment: type,
            },
          }
          showLoading()
          saveAlphabetAssessment({ variables })
            .then(() => {
              setLetters(defaultState)
              toggleModal()
              hideLoading()
            })
            .catch(() => {
              // TODO: An error message should be displayed to the user
              toggleModal()
              hideLoading()
            })
        }}
      >
        Save Progress
      </Button>
    </>
  )
}
