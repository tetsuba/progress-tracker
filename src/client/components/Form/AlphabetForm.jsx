import React, { useState } from 'react'
import { useMutation } from '@apollo/react-hooks'
import { Button } from 'react-bootstrap'

// CONTEXT
import { LoadingContext } from '../../context/LoadingContext'

// COMPONENTS
import Alphabet from '../Alphabet/Alphabet'

// QUERIES
import { GET_STUDENT_COURSE_ABC } from '../../api/course/courseABC/courseABC.query'

// MUTATIONS
import { SAVE_COURSE_PROGRESS_MUTATION } from '../../api/course/courseABC/courseABC.mutation'

// DEFAULT STATE
import ALPHABET_DEFAULT_STATE from '../Alphabet/Alphabet.defualtState'

// TODO: add loading indicator
type Props = {
  id: string,
}

export default function AlphabetForm(props: Props) {
  const { id } = props
  const { showLoading, hideLoading } = React.useContext(LoadingContext)
  const variables = {
    input: {
      id: id || '',
    },
  }

  const [letters, setLetters] = useState(ALPHABET_DEFAULT_STATE)
  const [saveCourseProgress] = useMutation(SAVE_COURSE_PROGRESS_MUTATION, {
    refetchQueries: [{ query: GET_STUDENT_COURSE_ABC, variables }],
  })

  return (
    <>
      <Alphabet letters={letters} update={setLetters} />
      <Button
        className="float-right"
        onClick={() => {
          const variables = {
            input: {
              id: id,
              date: String(Date.now()),
              alphabet: letters,
            },
          }
          showLoading()
          saveCourseProgress({ variables })
            .then(() => {
              setLetters(ALPHABET_DEFAULT_STATE)
              hideLoading()
            })
            .catch(() => hideLoading())
        }}
      >
        Save Progress
      </Button>
    </>
  )
}
