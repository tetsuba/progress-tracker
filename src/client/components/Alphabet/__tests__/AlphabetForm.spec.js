import { act } from 'react-dom/test-utils'
import { delay, graphRenderer } from '../../../../test/testHelper'
import AlphabetForm from '../AlphabetForm'
import { saveAlphabetAssessmentMutationSuccess } from '../../../../test/mockApi/assessment/alphabet/alphabetMockMutation'
import { getAlphabetAssessmentQuerySuccess } from '../../../../test/mockApi/assessment/alphabet/alphabetMockQuery'

describe('<AlphabetForm>', () => {
  const mockData = {
    getAlphabetAssessment: {
      history: [],
    },
  }

  const props = {
    type: 'upperCase',
    id: 'userId001',
    toggleModal: jest.fn(),
    data: mockData,
  }

  it('Should render a default state', async () => {
    let wrapper
    await act(async () => {
      wrapper = graphRenderer(
        AlphabetForm,
        [
          saveAlphabetAssessmentMutationSuccess,
          getAlphabetAssessmentQuerySuccess,
        ],
        props
      )
      await delay()
    })
    wrapper.update()
    expect(wrapper.find(AlphabetForm)).toMatchSnapshot()
  })

  it('Should close the modal when submitting an assessment', async () => {
    let wrapper
    await act(async () => {
      wrapper = graphRenderer(
        AlphabetForm,
        [
          saveAlphabetAssessmentMutationSuccess,
          getAlphabetAssessmentQuerySuccess,
        ],
        props
      )
      await delay()
    })
    wrapper.update()
    await act(async () => {
      wrapper.find('Button').props().onClick()
      await delay()
    })
    wrapper.update()
    expect(props.toggleModal).toHaveBeenCalledTimes(1)
  })
})
