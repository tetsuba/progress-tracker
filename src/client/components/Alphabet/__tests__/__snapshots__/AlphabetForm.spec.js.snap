// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlphabetForm> Should render a default state 1`] = `
<AlphabetForm
  data={
    Object {
      "getAlphabetAssessment": Object {
        "history": Array [],
      },
    }
  }
  id="userId001"
  toggleModal={[MockFunction]}
  type="upperCase"
>
  <Alphabet
    letters={
      Array [
        Object {
          "letter": "A",
          "value": 0,
        },
        Object {
          "letter": "B",
          "value": 0,
        },
        Object {
          "letter": "C",
          "value": 0,
        },
        Object {
          "letter": "D",
          "value": 0,
        },
        Object {
          "letter": "E",
          "value": 0,
        },
        Object {
          "letter": "F",
          "value": 0,
        },
        Object {
          "letter": "G",
          "value": 0,
        },
        Object {
          "letter": "H",
          "value": 0,
        },
        Object {
          "letter": "I",
          "value": 0,
        },
        Object {
          "letter": "J",
          "value": 0,
        },
        Object {
          "letter": "K",
          "value": 0,
        },
        Object {
          "letter": "L",
          "value": 0,
        },
        Object {
          "letter": "M",
          "value": 0,
        },
        Object {
          "letter": "N",
          "value": 0,
        },
        Object {
          "letter": "O",
          "value": 0,
        },
        Object {
          "letter": "P",
          "value": 0,
        },
        Object {
          "letter": "Q",
          "value": 0,
        },
        Object {
          "letter": "R",
          "value": 0,
        },
        Object {
          "letter": "S",
          "value": 0,
        },
        Object {
          "letter": "T",
          "value": 0,
        },
        Object {
          "letter": "U",
          "value": 0,
        },
        Object {
          "letter": "V",
          "value": 0,
        },
        Object {
          "letter": "W",
          "value": 0,
        },
        Object {
          "letter": "X",
          "value": 0,
        },
        Object {
          "letter": "Y",
          "value": 0,
        },
        Object {
          "letter": "Z",
          "value": 0,
        },
      ]
    }
    update={[Function]}
  >
    <div
      style={
        Object {
          "alignItems": "left",
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <Letter
        key="A-letter"
        letter="A"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          A
        </div>
      </Letter>
      <Letter
        key="B-letter"
        letter="B"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          B
        </div>
      </Letter>
      <Letter
        key="C-letter"
        letter="C"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          C
        </div>
      </Letter>
      <Letter
        key="D-letter"
        letter="D"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          D
        </div>
      </Letter>
      <Letter
        key="E-letter"
        letter="E"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          E
        </div>
      </Letter>
      <Letter
        key="F-letter"
        letter="F"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          F
        </div>
      </Letter>
      <Letter
        key="G-letter"
        letter="G"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          G
        </div>
      </Letter>
      <Letter
        key="H-letter"
        letter="H"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          H
        </div>
      </Letter>
      <Letter
        key="I-letter"
        letter="I"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          I
        </div>
      </Letter>
      <Letter
        key="J-letter"
        letter="J"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          J
        </div>
      </Letter>
      <Letter
        key="K-letter"
        letter="K"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          K
        </div>
      </Letter>
      <Letter
        key="L-letter"
        letter="L"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          L
        </div>
      </Letter>
      <Letter
        key="M-letter"
        letter="M"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          M
        </div>
      </Letter>
      <Letter
        key="N-letter"
        letter="N"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          N
        </div>
      </Letter>
      <Letter
        key="O-letter"
        letter="O"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          O
        </div>
      </Letter>
      <Letter
        key="P-letter"
        letter="P"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          P
        </div>
      </Letter>
      <Letter
        key="Q-letter"
        letter="Q"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          Q
        </div>
      </Letter>
      <Letter
        key="R-letter"
        letter="R"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          R
        </div>
      </Letter>
      <Letter
        key="S-letter"
        letter="S"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          S
        </div>
      </Letter>
      <Letter
        key="T-letter"
        letter="T"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          T
        </div>
      </Letter>
      <Letter
        key="U-letter"
        letter="U"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          U
        </div>
      </Letter>
      <Letter
        key="V-letter"
        letter="V"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          V
        </div>
      </Letter>
      <Letter
        key="W-letter"
        letter="W"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          W
        </div>
      </Letter>
      <Letter
        key="X-letter"
        letter="X"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          X
        </div>
      </Letter>
      <Letter
        key="Y-letter"
        letter="Y"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          Y
        </div>
      </Letter>
      <Letter
        key="Z-letter"
        letter="Z"
        update={[Function]}
        value={0}
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "border": "solid 1px #000",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "42px",
              "height": "100px",
              "justifyContent": "center",
              "margin": "4px",
              "padding": "",
              "width": "80px",
            }
          }
        >
          Z
        </div>
      </Letter>
    </div>
  </Alphabet>
  <Button
    active={false}
    className="float-right"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    <button
      className="float-right btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Save Progress
    </button>
  </Button>
</AlphabetForm>
`;
